generators/pythonv2/codegen/src/ast/Method.ts

```diff
@@ -0,0 +1,135 @@
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { Type } from "./Type";
+import { CodeBlock } from "./CodeBlock";
+import { ClassReference } from "./ClassReference";
+import { Field } from "./Field";
+import { Parameter } from "./Parameter";
+
+export enum MethodType {
+    STATIC,
+    INSTANCE,
+    CLASS
+}
+
+export declare namespace Method {
+    interface Args {
+        /* The name of the method */
+        name: string;
+        /* The parameters of the method */
+        parameters: Parameter[];
+        /* The return type of the method */
+        return_?: Type;
+        /* The body of the method */
+        body?: CodeBlock;
+        /* The docstring for the method */
+        docstring?: string;
+        /* The type of the method, defaults to STATIC */
+        type?: MethodType;
+        /* The class this method belongs to, if any */
+        classReference?: ClassReference;
+        /* Any decorators to add to the method */
+        decorators?: string[];
```

Comment:
do decorators need to be imported often? I could see this being a `ClassReference`

Reply:
I'm going to remove decorators from the scope of this PR and add them in separately and more hollistically.

I'm thinking about adding a new top-level `Decorator` ast node and adding `addDecorator` methods to both the `Class` and `Method` ast nodes.